app/models/user.rb

```diff
@@ -1,38 +1,67 @@
 class User < ApplicationRecord
-  include Authenticatable
-  before_create :generate_auth_token
-  before_destroy :check_last_admin
+  ### CONSTANTS ###
 
-  enum role: {
-         user: 0,
-         admin: 1
-       }
+  ROLES = {
+    user: 0,
+    admin: 1
+  }
 
   ### VALIDATIONS ###
 
   validates :first_name, :last_name, :email, presence: true
   validates :email, uniqueness: true, email: true
 
+  ### CALLBACKS ###
+
+  before_create :generate_auth_token
```

Comment:
it should be like set_auth_token as it is also setting the value on user

Reply:
Done